File: ayt-01/03Collatz.c
#include <stdio.h>
void collatz(int n)
{
    printf("%i ", n);
    if(n == 1) return;
    else if(n & 1) collatz(3 * n + 1);
    else collatz(n >> 1);
}
int main(int argv, char **argc)
{
    collatz(3);
    return 0;
}

File: ayt-01/03Collatz.java
class Collatz {
    static void collatz(int n)
    {
        System.out.println(n);
        if(n == 1) return;
        else if((n & 1) == 1) collatz(((n << 1)|1) + n);
        else collatz(n >> 1);
    }
    public static void main(String[] args)
    {
        collatz(3);
    }
}